### ISSUE TITLE
Transaction statements endpoint returns multiple types of fees even with receiptType filter

### ISSUE THREAD

**Issue by evias** [nemtech/catapult-rest#452]
**Expected Behaviour**

When calling the `/statements/transaction?height=7405&receiptType=8515`, we expect the endpoint to return **only** receipts of **type 8515**. 

**Actual Behaviour**

 http://api-01.eu-central-1.096x.symboldev.network:3000/statements/transaction?height=7405&receiptType=8515

As seen in the endpoint above, there is also a receipt with type `20803` that is returned when specifying the receiptType `8515`.

**Comment by Vektrat** [comment]
the endpoint serves to retrieve `statements` which contain a list of receipts

the filter returns statements that contain at least one receipt with the provided type, if we were to further filter the inner receipts array, we would be breaking/trimming the statement object, which is undesired

thus this extra filtering falls on client hands - i think what you are referring to would be if the endpoint was `/receipts/STATEMENT_ID?type=XXX` which would instead, return the receipts a statement contains, and those could be filtered

**Comment by evias** [comment]
What does the receiptType filter do, if not to restrict which type of receipts will be returned?

The endpoint i mentioned should return transaction statements, indeed, but the fact that there is a "receiptType" url parameter possibility gives sense to filtering those receipts wouldnt you think, also?

----- Message de réponse -----
De : "Roger Hernandez" <notifications@github.com>
Pour : "nemtech/catapult-rest" <catapult-rest@noreply.github.com>
Cc : "Grégory Saive" <greg@evias.be>, "Author" <author@noreply.github.com>
Objet : [nemtech/catapult-rest] Transaction statements endpoint returns multiple types of fees even with receiptType filter (#452)
Date : lun., août 24, 2020 10:25

the endpoint serves to retrieve statements which contain a list of receipts
the filter returns statements that contain at least one receipt with the provided type, if we were to further filter the inner receipts array, we would be breaking/trimming the statement object, which is undesired
thus this extra filtering falls on client hands - i think what you are referring to would be if the endpoint was /receipts/STATEMENT_ID?type=XXX which would instead, return the receipts a statement contains, and those could be filtered

&mdash;You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by Vektrat** [comment]
no, i don't think those should be filtered, because then you'd be modifying the objects that are being returned, and the client could think they are complete, when in fact they are being tampered with

what it does is: returns statements that have at least one receipt with the given type, just like `/accounts` with `mosaicId`, it filters accounts that have some amount of the given `mosaicId`, but it does not remove the other owned mosaics from the results
